.eslintrc.json

```diff
@@ -0,0 +1,294 @@
+		"array-element-newline":        ["warn", "consistent"],
+		"arrow-parens":                 ["warn", "as-needed"],
+		"arrow-spacing":                ["error", {"before": true, "after": true}],
+		"block-spacing":                ["error", "always"],
+		"brace-style":                  ["error", "1tbs", {"allowSingleLine": true}],
+		"comma-dangle":                 ["warn", "never"],
+		"comma-spacing":                ["error", {"before": false, "after": true}],
+		"comma-style":                  ["error", "last"],
+		"computed-property-spacing":    ["error", "never"],
+		"dot-location":                 ["error", "property"],
+		"eol-last":                     ["error", "always"],
+		"func-call-spacing":            ["error", "never"],
+		"function-call-argument-newline": ["error", "consistent"],
+		"function-paren-newline":       ["error", "never"],
+		"generator-star-spacing":       ["warn", {"before": false, "after": true}],
+		"implicit-arrow-linebreak":     ["error", "beside"],
+		"indent":                       ["error", "tab"],
+		"jsx-quotes":                   ["warn", "prefer-double"],
+		"key-spacing":                  ["warn", {
+			"beforeColon": false, "afterColon": true,
+			"mode": "minimum",    "align": "value"
+		}],
+		"keyword-spacing":              ["error", {"before": true, "after": true}],
+		"line-comment-position":         "off",
+		"linebreak-style":              ["error", "unix"],
+		"lines-around-comment":         ["warn", {
+			"beforeBlockComment": true,
+			"beforeLineComment":  true,
+			"allowBlockStart":    true,
+			"allowBlockEnd":      false,
+			"allowObjectStart":   true,
+			"allowObjectEnd":     false,
+			"allowArrayStart":    true,
+			"allowArrayEnd":      false,
+			"allowClassStart":    false,
+			"allowClassEnd":      false
+		}],
+		"lines-between-class-members":  ["error", "always", {"exceptAfterSingleLine": true}],
+		"max-len":                      ["warn", {
+			"code": 160, "tabWidth": 2, "comments": 120, "ignoreUrls": true
+		}],
+		"max-statements-per-line":      ["error", {"max": 1}],
+		"multiline-ternary":            ["warn", "never"],
+		"new-parens":                   ["error", "always"],
+		"newline-per-chained-call":     ["error", {"ignoreChainWithDepth": 3}],
+		"no-extra-parens":               "off",
+		"no-mixed-spaces-and-tabs":     ["error", "smart-tabs"],
+		"no-trailing-spaces":            "error",
+		"no-whitespace-before-property": "error",
+		"nonblock-statement-body-position": ["error", "beside"],
+		"object-curly-newline":         ["error", {"multiline": true}],
+		"object-curly-spacing":         ["warn", "always"],
+		"object-property-newline":      ["error", {"allowAllPropertiesOnSameLine": true}],
+		"operator-linebreak":           ["error", "after"],
+		"padded-blocks":                ["warn", {"blocks": "never", "classes": "always", "switches": "never"}, {"allowSingleLineBlocks": true}],
+		"padding-line-between-statements": ["warn",
```

Comment:
```suggestion
		"padding-line-between-statements": ["off",
```

Reply:
objection rejected.
If you see anything were there shouldn't be a padding line, tell me, we can tweak the settings

Reply:
![image](https://user-images.githubusercontent.com/44261356/236631623-b0f06ba7-c2ef-4b75-812b-8dac6a0cdef5.png)
that warning there is literally stupid.

Reply:
I'm gonna add it there too, don't worry.
My code isn't perfect either

Reply:
no, its stupid to need an empty line there

Reply:
Defining constants and return statements should be visually separated to increase clarity.
Also, are you really gonna complain that *I* have to change something in *my* code? xD

Reply:
?

Reply:
It's just an empty line to improve readability. What's the problem with that?

Reply:
Its stupid

Reply:
Again, it improves code readability, even if you don't see it.
There is a difference between:
```ts
function foo(c: number) {
  const a = 5
  const b = 6
  const ab = 5+6
  const abc = ab + c
  if (ab > abc) {
    for (let i = 0; i < abc; i++) {
      console.log(a + i)
      if (a + i > b) return c * i
    }
  }
 let e = 2
 while (e > ab - c) {
   if (e < 3) return e+2
   else if (e > abc) e--
   return 21
 }
 return a + " and " + ab + " but " + e
}
```
and
```ts
function foo(c: number) {
  const a = 5
  const b = 6
  const ab = 5 + 6
  const abc = ab + c
  
  if (ab > abc) {
    for (let i = 0; i < abc; i++) {
      console.log(a + i)
      
      if (a + i > b) return c * i
    }
  }
  
 let e = 2
 while (e > ab - c) {
   if (e < 3) return e + 2
   else if (e > abc) e--
   
   return 21
 }
 
 return a + " and " + ab + " but " + e
}
```
I know, both are nonsense code, but it should suffice to get the point across.

Reply:
It's also automatic on save if you have the vscode extension for eslint, so it's effortless.